logstash-core/lib/logstash/api/rack_app.rb

```diff
@@ -8,9 +8,84 @@
 module LogStash
   module Api
     module RackApp
-      def self.app
+      # Cabin is not compatible with CommonLogger, and this gives us more control anyway
+      class ApiLogger
+        LOG_MESSAGE = "API HTTP Request".freeze
+        
+        def initialize(app, logger)
+          @app = app
+          @logger = logger
+        end
+
+        def call(env)
+          res = @app.call(env)
+          status, headers, body = res
+          
+          if status >= 500 && status < 600
+            @logger.warn? && @logger.warn(LOG_MESSAGE, log_metadata(status, env))                      
+          else          
+            @logger.info? && @logger.info(LOG_MESSAGE, log_metadata(status, env))                      
+          end
+
+          res          
+        end
+
+        def log_metadata(status, env)
+          {
+            :status => status,
+            :method => env["REQUEST_METHOD"],
+            :path => env["REQUEST_PATH"],
+            :query_string => env["QUERY_STRING"],
+            :http_version => env["HTTP_VERSION"],
+            :http_accept => env["HTTP_ACCEPT"]
+          }
+        end
+          
+      end
+      
+      class ApiErrorHandler
```

Comment:
why having two different handlers? one for http errors and another for logging exceptions?